java/test/jmri/jmrit/logix/LearnWarrantTest.java

```diff
@@ -96,24 +84,18 @@ public void testLearnWarrant() throws Exception {
         Sensor sensor = block0.getSensor();
         NXFrameTest.setAndConfirmSensorAction(sensor, Sensor.ACTIVE, block0);
 
-        JUnitUtil.waitFor(() -> {
-            return  (block0.getState() & OBlock.ALLOCATED | OBlock.OCCUPIED) != 0;
```

Comment:
I just wanted to point out the subtle bug in this line.

The & happens before the |, so the result of the & doesn’t matter because the | with a constant always makes the value nonzero.

The solution is to put the | in parentheses. I moved that line to the oBlovkOccupiedOrAllocated method.